server/app/models.py

```diff
@@ -180,6 +180,34 @@ def get_final_submission(self, assignment_key):
                 FinalSubmission.assignment==assignment_key,
                 FinalSubmission.submitter==self.key).get()
 
+    def get_submission_before(self, assignment_key, before_time):
+        """Get this users last submission before some server time"""
+        if isinstance(assignment_key, Assignment):
+            assignment_key = assignment_key.key
```

Comment:
There's a little bit of code duplication here (between the normal `get_submission_before`